spec/requests/users_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
+  describe 'GET/index' do
+    it 'should return http success for users' do
+      get users_url
+      expect(response).to be_successful
+      expect(response).to render_template(:index) # should ensure that the correct template is rendered
+      assert_equal 'index', @controller.action_name # should ensure the right action is rendered
+      # should correct body for the post view file is rendered
+      assert_match 'Here is the list of users for the rails blog app', @response.body
+    end
+  end
+
+  describe 'GET/index/:id' do
+    it 'should return http success for user' do
+      get user_url(User.last)
+      expect(response).to be_successful
+      expect(response).to render_template(:show) # should ensure that the correct template is rendered
+      assert_equal 'show', @controller.action_name # should ensure the right action is rendered
+      # should correct body for the post view file is rendered
+      assert_match 'Shows a specific user', @response.body
+    end
+  end
+
+  describe 'GET/index/:id/posts' do
+    it 'should return http success for user`s posts' do
+      get user_posts_url(Post.last)
+      expect(response).to be_successful
+      expect(response).to render_template(:index) # should ensure that the correct template is rendered
+      assert_equal 'index', @controller.action_name # should ensure the right action is rendered
+      # should correct body for the post view file is redered
+      assert_match 'Displays a list of posts from a user', @response.body
+    end
+  end
+
+  describe 'GET/index/:id/:id/posts/:id' do
+    it 'should return http success for user`s post' do
+      get user_post_url(user_id: User.last, id: Post.last.id)
+      expect(response).to be_successful
+      expect(response).to render_template(:show) # should ensure that the correct template is rendered
+      assert_equal 'show', @controller.action_name # should ensure the right action is rendered
+      # should correct body for the post view file is redered
+      assert_match 'Shows a single post from a user ...', @response.body
+    end
+  end
```

Comment:
- [ ] These are not users controller specs. Please make a new file called `posts_spec.rb` and include them there.
- [ ] It would be better professionally if you made every test inside `it` method like this:

```ruby
describe 'GET /index' do
    before :each do
       get user_posts_url(Post.last)
    end
    it 'should return http success for user`s posts' do
      expect(response).to be_successful 
    end
   it 'should ensure that the correct template is rendered' do
      expect(response).to render_template(:index)
    end
   
   .........
   .........
end
```